### PR TITLE
arm_cpu_test_clusters:Check guest vcpu clusters

### PR DESCRIPTION
**Pull request by zhenyzha**
Check guest gets correct multiple vcpu clusters

ID:2093857
Signed-off-by: zhenyzha <zhenyzha@redhat.com>

### CODE REVIEW

**Comment by zhenyzha** [comment]
This patch depends on:
https://github.com/avocado-framework/avocado-vt/pull/3416

File: qemu/tests/arm_cpu_test_clusters.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    vm_name = params['main_vm']
+    vcpu_clusters_list = [2, 4]
+    params['vcpu_clusters'] = random.choice(vcpu_clusters_list)
+    params['start_vm'] = 'yes'
+    env_process.preprocess_vm(test, params, env, vm_name)
```
Comment (PaulYuuu) [inline qemu/tests/arm_cpu_test_clusters.py:23]: Seems no contains some prepare steps, I think you can use `vm.create(params=params)` directly.

File: qemu/tests/arm_cpu_test_clusters.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    vm = env.get_vm(vm_name)
+    session = vm.wait_for_login()
+    if params["os_type"] == "linux":
```
Comment (PaulYuuu) [inline qemu/tests/arm_cpu_test_clusters.py:26]: For now, this is meaningless.

File: qemu/tests/arm_cpu_test_clusters.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        check_cluster_id = params['check_cluster_id']
+        check_cluster_cpus_list = params['check_cluster_cpus_list']
+        vcpu_sockets = vm.cpuinfo.sockets
+        vcpu_clusters = vm.cpuinfo.clusters
+        clusters_id = session.cmd_output(check_cluster_id).strip().split('\n')
```
Comment (PaulYuuu) [inline qemu/tests/arm_cpu_test_clusters.py:31]:
```suggestion
        clusters_id = session.cmd_output(check_cluster_id).strip().splitlines()
```

File: qemu/tests/cfg/arm_cpu_test_clusters.cfg [hunk 4]
```diff
@@ -0,0 +1,14 @@
+- arm_cpu_test_clusters:
+    type = arm_cpu_test_clusters
+    kill_vm_on_error = yes
+    start_vm = no
+    check_host_flags = yes
```
Comment (PaulYuuu) [inline qemu/tests/cfg/arm_cpu_test_clusters.cfg:5]: Unuse.

File: qemu/tests/arm_cpu_test_clusters.py (hunk 1 above)
Comment (zhenyzha) [reply qemu/tests/arm_cpu_test_clusters.py]: corrected

File: qemu/tests/arm_cpu_test_clusters.py (hunk 3 above)
Comment (zhenyzha) [reply qemu/tests/arm_cpu_test_clusters.py]: corrected

File: qemu/tests/cfg/arm_cpu_test_clusters.cfg (hunk 4 above)
Comment (zhenyzha) [reply qemu/tests/cfg/arm_cpu_test_clusters.cfg]: corrected

File: qemu/tests/arm_cpu_test_clusters.py (hunk 2 above)
Comment (zhenyzha) [reply qemu/tests/arm_cpu_test_clusters.py]: corrected

**Comment by zhenyzha** [comment]
(1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.arm_cpu_test_clusters.arm64-pci: PASS (50.99 s)

@PaulYuuu Could you pls help review again when you are free. thanks.

File: qemu/tests/arm_cpu_test_clusters.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    vm.verify_alive()
+    session = vm.wait_for_login()
+
+    check_cluster_id = params['check_cluster_id']
+    check_cluster_cpus_list = params['check_cluster_cpus_list']
+    vcpu_sockets = vm.cpuinfo.sockets
+    vcpu_clusters = vm.cpuinfo.clusters
+    clusters_id = session.cmd_output(check_cluster_id).strip().splitlines()
+    clusters_cpus_list = session.cmd_output(
+        check_cluster_cpus_list).strip().split('\n')
```
[the remark this answers is not in the document]
Comment (zhenyzha) [reply qemu/tests/arm_cpu_test_clusters.py]: Sorry, it has been corrected.

File: qemu/tests/cfg/arm_cpu_test_clusters.cfg [hunk 6]
```diff
@@ -0,0 +1,19 @@
+- arm_cpu_test_clusters:
+    type = arm_cpu_test_clusters
+    kill_vm_on_error = yes
+    start_vm = no
+    reboot_method = "shell"
```
[the remark this answers is not in the document]
Comment (zhenyzha) [reply qemu/tests/cfg/arm_cpu_test_clusters.cfg]: corrected

**Comment by zhenyzha** [comment]
git am 3346.patch
Applying: arm_cpu_test_clusters:Check guest vcpu clusters

` (1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.arm_cpu_test_clusters.arm64-pci: PASS (82.29 s)`

@PaulYuuu Could you pls help review again when you are free. thanks.

File: qemu/tests/cfg/arm_cpu_test_clusters.cfg [hunk 7]
```diff
@@ -0,0 +1,18 @@
+- arm_cpu_test_clusters:
+    type = arm_cpu_test_clusters
+    kill_vm_on_error = yes
+    start_vm = no
+    only aarch64
+    # support on RHEL.9.1
+    no RHEL.8 RHEL.9.0
+    required_qemu = [7.0.0, )
+    vcpu_maxcpus = 0
```
Comment (PaulYuuu) [inline qemu/tests/cfg/arm_cpu_test_clusters.cfg:9]: make smp=vcpu_maxcpus=0 safer, to avoid users defining SMP by themselves. Other LGTM.

Comment (zhenyzha) [reply qemu/tests/cfg/arm_cpu_test_clusters.cfg:9]: good suggestion

Comment (zhenyzha) [reply qemu/tests/cfg/arm_cpu_test_clusters.cfg:10]: corrected

**Comment by zhenyzha** [comment]
(1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.arm_cpu_test_clusters.arm64-pci: PASS (61.52 s)

@PaulYuuu Could you pls help review again when you are free. thanks.

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**